homeassistant/components/ecoforest/switch.py

```diff
@@ -0,0 +1,75 @@
+"""Switch platform for Ecoforest."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyecoforest.models.device import Device
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+from .entity import EcoforestEntity
+
+
+@dataclass
+class EcoforestRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Device], bool]
+
+
+@dataclass
+class EcoforestSwitchEntityDescription(
+    SwitchEntityDescription, EcoforestRequiredKeysMixin
+):
+    """Describes an Ecoforest switch entity."""
+
+
+SWITCH_TYPES: tuple[EcoforestSwitchEntityDescription, ...] = (
+    EcoforestSwitchEntityDescription(
+        key="status", translation_key="status", value_fn=lambda data: data.on
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Ecoforest switch platform."""
+    coordinator: EcoforestCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        EcoforestSwitchEntity(coordinator, description) for description in SWITCH_TYPES
+    ]
+
+    async_add_entities(entities)
+
+
+class EcoforestSwitchEntity(EcoforestEntity, SwitchEntity):
+    """Representation of an Ecoforest switch entity."""
+
+    entity_description: EcoforestSwitchEntityDescription
+
+    _attr_name = None
+
```

Comment:
```suggestion
```
I would move this to the entity description

Reply:
Didn't knew that was possible. 👍